### PR TITLE
ADBDEV-288

### PR DESCRIPTION
**Pull request by darthunix**
At first the problem was detected in Greenplum
```
create table ta(a int check(a = 1));
insert into ta values(null);

set optimizer=off;
select * from ta where a is null;
 a 
---
  
(1 row)

set optimizer=on;
select * from ta where a is null;
 a 
---
(0 rows)

set optimizer_print_query = on;
set client_min_messages='log';
select * from ta where a is null;
Algebrized query: 
+--CLogicalSelect
   |--CLogicalGet "ta" ("ta"), Columns: ["a" (0), "ctid" (1), "xmin" (2), "cmin" (3), "xmax" (4), "cmax" (5), "tableoid" (6), "gp_segment_id" (7)] Key sets: {[1,7]}
   +--CScalarNullTest
      +--CScalarIdent "a" (0)

Algebrized preprocessed query: 
+--CLogicalConstTableGet Columns: ["a" (0), "ctid" (1), "xmin" (2), "cmin" (3), "xmax" (4), "cmax" (5), "tableoid" (6), "gp_segment_id" (7)] Values: []
```
So ORCA preprocesses a logical plan to a single node  CLogicalConstTableGet, that means "there is no suitable data in this node, no need for retrieving data and constructing a physical plan". But in fact it is a wrong behavior for `ta` table - it can contain nulls as far as check `a = 1` doesn't protect from nulls. `select null = 1` returns null, but not a boolean value.

It was easy to guess that the problem was connected with constraint null checks and comparing with constants. Debugging showed out that CLogicalConstTableGet node is formed in [CExpressionPreprocessor.cpp](https://github.com/greenplum-db/gporca/blob/v3.65.2/libgpopt/src/operators/CExpressionPreprocessor.cpp#L1644) file and in a current example max cardinality is zero.

So here is a pipeline that caused the problem: [CLogicalSelect.cpp](https://github.com/greenplum-db/gporca/blob/v3.65.2/libgpopt/src/operators/CLogicalSelect.cpp#L141) -> [CConstraintInterval.cpp](https://github.com/greenplum-db/gporca/blob/v3.65.2/libgpopt/src/base/CConstraintInterval.cpp#L83)

As you can see the problem is in `m_fIncludesNull` parameter that describes whether nulls are allowed in a constraint or not. At the moment it is always false. But in fact it should be false only if a relative column not null - otherwise it should be true.

Under hood `m_fIncludesNull` is set in a function  [PciIntervalFromColConstCmp](https://github.com/greenplum-db/gporca/blob/v3.65.2/libgpopt/src/base/CConstraintInterval.cpp#L391) that transforms constraints comparing a column with a  constant to an interval one. So `a = 1` constraint always transforms to `a in [1, 1]` and `m_fIncludesNull` is always set false at the moment for a new interval constraint.

As a solution I propose to set `m_fIncludesNull` depending on whether a relative column is nullable or not. I am not sure that it solves the root of a problem but it seems to be a small change in a code and may be a good starting point.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ project(gpopt LANGUAGES CXX C)
 set(CMAKE_CXX_STANDARD 98)
 set(GPORCA_VERSION_MAJOR 3)
 set(GPORCA_VERSION_MINOR 65)
-set(GPORCA_VERSION_PATCH 0)
+set(GPORCA_VERSION_PATCH 3)
```
Comment (darthunix) [inline CMakeLists.txt]: I am not sure that I have to bump a version - I can amend this change

**Review by leskin-in — APPROVED** [review]

**Review by leskin-in — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (darthunix) [reply CMakeLists.txt]: I have removed changes in gporca patch version - they are not relative to this PR and should be done only if Pivotal would ask for it

**Comment by leskin-in** [comment]
Could you explain the origin and usage of dumps which are created by this PR?

**Review by dursegov — APPROVED** [review]

**Comment by darthunix** [comment]
> Could you explain the origin and usage of dumps which are created by this PR?

Yes, it is my fault that I haven't described them it PR (I should do it for upstream PR). Minidump files is a standard way the create tests in ORCA. They contain an incoming query from GP (or other system) in DXL format: all information about affected relations, input/output columns, constraints etc + an output plan in DXL. So minidump file is enough to compare the output plan generated by ORCA on input section of minidump with its output section.

My goal was to test ORCA behaviour of two scenarios - a table with a constant compare check for a nullable column and for a non-nullable column.
```
create table ta(a int check(a = 1));
insert into ta values(null);

create table tb(b int not null check(b = 1));
insert into tb values(1);
```
`ConstraintIntervatIncludesNull.mdp` was generated with a query that returned one null row with a TableScan: `select * from ta where a is null;` 

`ConstraintIntervalNotIncludesNull.mdp` was generated with a query that returned an empty output using OneTimeFilter (don't try to scan a table, there is nothing suitable there): `select * from tb where b is null;` 

More information about generating minidumps is [here](https://github.com/greenplum-db/gporca#addtest).

**Review by Stolb27 — APPROVED** [review]

**Pull request closed without merging**